### PR TITLE
Implementation of Destination Search History feature

### PR DESCRIPTION
**Pull request by abhandal**
Addresses #70 implemented by #56 

This feature allows the user to view their destination search history, remove items from their history, and select an item from their history to view their corresponding landmarks and save the new destination within a trip. 

The PR includes the features and its respective unit and espresso UI tests.

### CODE REVIEW

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 1]
```diff
@@ -50,19 +50,22 @@
...
+    Button bSave;
+    @BindView(R.id.planner_title)
+    TextView tvTitle;
 
     private PlannerFragmentPagerAdapter adapter;
     private Trip openTrip;
     private Unbinder unbinder;
-    private List<Trip> userTripsList = new ArrayList<>();
+    private List <Trip> userTripsList = new ArrayList <>();
+    private List <Trip> userDestinationList = new ArrayList <>();
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

**Comment by amawai** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 29
- Added 20
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985&bid=7015725)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 2]
```diff
@@ -118,17 +121,17 @@ private AppCompatActivity getAppCompatActivity() {
...
     }
 
+    /*
+        Child fragment callbacks
+     */
+
     @Override
     public void onNewTrip() {
-        CallbackTask.execute(() -> TripDbHelper.instance().createList(), new CallbackTask.DefaultCallback<Trip>(){
+        CallbackTask.execute(() -> TripDbHelper.instance().createList(), new CallbackTask.DefaultCallback <Trip>() {
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 3]
```diff
@@ -242,10 +256,41 @@ public void failure(Throwable caught) {
...
+            }
+        });
+    }
+
     public void updateUserTripList() {
-        CallbackTask.execute(() -> TripDbHelper.instance().getAllLists(), new CallbackTask.DefaultCallback<List<Trip>>() {
+        CallbackTask.execute(() -> TripDbHelper.instance().getAllLists(), new CallbackTask.DefaultCallback <List <Trip>>() {
             @Override
-            public void success(List<Trip> list) {
+            public void success(List <Trip> list) {
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 4]
```diff
@@ -183,16 +191,22 @@ public void onShareTrip(int index) {
...
+    }
 
     @Override
     public void onDateChanged(int year, int month, int day) {
         openTrip.setTripDepartureDate(year, month, day);
     }
 
     //saves landmarks into trips, called in onclick
-    public void onSave(List<LandmarkCard> saveList) {
+    public void onSave(List <LandmarkCard> saveList) {
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 5]
```diff
@@ -242,10 +256,41 @@ public void failure(Throwable caught) {
...
+            @Override
+            public void failure(Throwable caught) {
+                Toast.makeText(getActivity(), "Failed to delete the destination", Toast.LENGTH_SHORT).show();
+            }
+        });
+    }
+
     public void updateUserTripList() {
-        CallbackTask.execute(() -> TripDbHelper.instance().getAllLists(), new CallbackTask.DefaultCallback<List<Trip>>() {
+        CallbackTask.execute(() -> TripDbHelper.instance().getAllLists(), new CallbackTask.DefaultCallback <List <Trip>>() {
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/trip/Trip.java [hunk 6]
```diff
@@ -99,13 +114,11 @@ public Date getTripDepartureDate() {
...
     public void setTripDepartureDate(int year, int month, int date) {
-        if (departureDate != null)
-        {
+        if (departureDate != null) {
             this.departureDate.setYear(year);
             this.departureDate.setMonth(month);
             this.departureDate.setDate(date);
-        }
-        else
+        } else
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/trip/Trip.java]: Codacy Issue found: ['else' construct must use '{}'s.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 7]
```diff
@@ -255,15 +300,28 @@ public void success(List<Trip> list) {
         });
     }
 
+    public void updateUserDestinationList() {
+        CallbackTask.execute(() -> DestinationDbHelper.getInstance().getAllLists(), new CallbackTask.DefaultCallback <List <Trip>>() {
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/destinationpicker/DestinationFragment.java [hunk 8]
```diff
@@ -18,32 +24,36 @@
...
 
+import static android.widget.LinearLayout.VERTICAL;
+
 /**
  * Created by abhandal on 3/3/2018.
  */
 
 public class DestinationFragment extends Fragment {
 
+    @BindView(R.id.destination_history_view_recycler)
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/destinationpicker/DestinationFragment.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/destinationpicker/DestinationFragment.java [hunk 9]
```diff
@@ -88,4 +141,100 @@ private AppCompatActivity getAppCompatActivity() {
...
+        // Insert a new item to the RecyclerView on a predefined position, could be used in the future
+        public void insert(int position, Trip data) {
+            userDestinationList.add(position, data);
+            notifyItemInserted(position);
+        }
+    }
+
+    public final class DestinationItemHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+
+        @BindView(R.id.destination_view_list)
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/destinationpicker/DestinationFragment.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/test/java/org/wikipedia/travel/database/DestinationDbHelperTest.java [hunk 10]
```diff
@@ -0,0 +1,60 @@
+package org.wikipedia.travel.database;
```
Comment (amawai) [inline app/src/test/java/org/wikipedia/travel/database/DestinationDbHelperTest.java]: Codacy Issue found: [File does not end with a newline.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/androidTest/java/org/wikipedia/espresso/travel/DestinationHistoryEspressoTest.java [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+        mDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation());
+
+        //Navigate to the Destinations fragment
+        ViewInteraction planNewTripButton = onView(
+                allOf(withId(R.id.trip_button_new)));
+        planNewTripButton.perform(click());
+    }
+
+    @Test
+    public void selectedDestinationShouldAppearInDestinationHistory() throws UiObjectNotFoundException {
```
Comment (amawai) [inline app/src/androidTest/java/org/wikipedia/espresso/travel/DestinationHistoryEspressoTest.java]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 12]
```diff
@@ -50,19 +50,22 @@
...
+    @BindView(R.id.planner_save)
+    Button bSave;
+    @BindView(R.id.planner_title)
+    TextView tvTitle;
 
     private PlannerFragmentPagerAdapter adapter;
     private Trip openTrip;
     private Unbinder unbinder;
-    private List<Trip> userTripsList = new ArrayList<>();
+    private List <Trip> userTripsList = new ArrayList <>();
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 13]
```diff
@@ -225,11 +239,11 @@ public boolean onBackPressed() {
      */
     private void saveTrip(Trip trip) {
         bSave.setActivated(false);
-        CallbackTask.execute(() -> TripDbHelper.instance().updateList(trip),  new CallbackTask.DefaultCallback<Object>(){
+        CallbackTask.execute(() -> TripDbHelper.instance().updateList(trip), new CallbackTask.DefaultCallback <Object>() {
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 14]
```diff
@@ -50,19 +50,22 @@
...
-    public interface Callback {
-        void onLoadPage(PageTitle title, HistoryEntry entry);
-    }
-    @BindView(R.id.fragment_travel_planner_view_pager) ViewPager viewPager;
-    @BindView(R.id.planner_next) Button bNext;
-    @BindView(R.id.planner_save) Button bSave;
-    @BindView(R.id.planner_title) TextView tvTitle;
+        DestinationFragment.Callback, DateFragment.Callback, LandmarkFragment.Callback {
+
+    @BindView(R.id.fragment_travel_planner_view_pager)
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/destinationpicker/DestinationFragment.java [hunk 15]
```diff
@@ -88,4 +141,100 @@ private AppCompatActivity getAppCompatActivity() {
...
+        RelativeLayout destinationLayout;
+        @BindView(R.id.destination_name_view_text)
+        TextView destinationName;
+        @BindView(R.id.destination_item_icon)
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/destinationpicker/DestinationFragment.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 16]
```diff
@@ -242,10 +256,41 @@ public void failure(Throwable caught) {
         });
     }
 
+    private void saveDestination() {
+        CallbackTask.execute(() -> DestinationDbHelper.getInstance().createList(openTrip.getDestination()), new CallbackTask.DefaultCallback <Object>() {
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 17]
```diff
@@ -154,10 +158,14 @@ public void onRequestTripListUpdate() {
...
 
+    @Override
+    public void onRequestDestinationListUpdate() {
+        updateUserDestinationList();
+    }
 
     @Override
     public void onDeleteTrip(long id) {
-        CallbackTask.execute(() -> TripDbHelper.instance().deleteList(getTrip(id)), new CallbackTask.DefaultCallback<Object>() {
+        CallbackTask.execute(() -> TripDbHelper.instance().deleteList(getTrip(id)), new CallbackTask.DefaultCallback <Object>() {
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 18]
```diff
@@ -242,10 +256,41 @@ public void failure(Throwable caught) {
...
+            @Override
+            public void failure(Throwable caught) {
+                super.failure(caught);
+            }
+        });
+    }
+
+    @Override
+    public void onDeleteDestination(long id) {
+        CallbackTask.execute(() -> DestinationDbHelper.getInstance().deleteList(userDestinationList.get((int) id)), new CallbackTask.DefaultCallback <Object>() {
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: ['<' is preceded with whitespace.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 19]
```diff
@@ -50,19 +50,22 @@
...
-    @BindView(R.id.planner_title) TextView tvTitle;
+        DestinationFragment.Callback, DateFragment.Callback, LandmarkFragment.Callback {
+
+    @BindView(R.id.fragment_travel_planner_view_pager)
+    ViewPager viewPager;
+    @BindView(R.id.planner_next)
+    Button bNext;
+    @BindView(R.id.planner_save)
+    Button bSave;
+    @BindView(R.id.planner_title)
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/main/java/org/wikipedia/travel/destinationpicker/DestinationFragment.java [hunk 20]
```diff
@@ -88,4 +141,100 @@ private AppCompatActivity getAppCompatActivity() {
...
+            userDestinationList.add(position, data);
+            notifyItemInserted(position);
+        }
+    }
+
+    public final class DestinationItemHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+
+        @BindView(R.id.destination_view_list)
+        RelativeLayout destinationLayout;
+        @BindView(R.id.destination_name_view_text)
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/destinationpicker/DestinationFragment.java]: Codacy Issue found: [Use explicit scoping instead of the default package private level](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

File: app/src/test/java/org/wikipedia/travel/database/DestinationDbHelperTest.java (hunk 10 above)
Comment (amawai) [inline app/src/test/java/org/wikipedia/travel/database/DestinationDbHelperTest.java]: Codacy Issue found: [File does not end with a newline.](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985)

**Comment by amawai** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 30
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985&bid=7015725)

**Comment by amawai** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 30
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985&bid=7015725)

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 21]
```diff
@@ -242,6 +256,37 @@ public void failure(Throwable caught) {
         });
     }
 
+    private void saveDestination() {
```
Comment (jadmalek) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: For the new destination history table, are the destinations stored only with the destination name-based constructor? To clarify, the search history displayed will be the same when you edit any of the trips, right?

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java [hunk 22]
```diff
@@ -255,15 +300,28 @@ public void success(List<Trip> list) {
         });
     }
 
+    public void updateUserDestinationList() {
```
Comment (jadmalek) [inline app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: Can you clarify what exactly happens when this method is invoked?

File: app/src/test/java/org/wikipedia/travel/database/DestinationDbHelperTest.java [hunk 23]
```diff
@@ -0,0 +1,60 @@
...
+import static org.mockito.Mockito.mock;
+
+public class DestinationDbHelperTest {
+    private SQLiteDatabase mockDb;
+    private DestinationDbHelper mockDbHelper;
+    private Trip.Destination mockDestination;
+
+    //makes sure each test starts with an uninitialized singleton and your tests are no longer affecting each other.
+    @Before
+    public void resetSingleton() throws SecurityException, NoSuchFieldException, IllegalArgumentException, IllegalAccessException {
```
Comment (jadmalek) [inline app/src/test/java/org/wikipedia/travel/database/DestinationDbHelperTest.java]: What is the purpose of this method?

File: app/src/test/java/org/wikipedia/travel/database/DestinationDbHelperTest.java [hunk 24]
```diff
@@ -0,0 +1,60 @@
...
+        assertNotNull(DestinationDbHelper.getInstance());
+    }
+
+    @Test
+    public void mockTestGetAllLists() {
+        mockDbHelper.getAllLists();
+        assertEquals(0, mockDbHelper.getAllLists().size());
+    }
+
+    @Test
```
Comment (jadmalek) [inline app/src/test/java/org/wikipedia/travel/database/DestinationDbHelperTest.java]: Would be possible to do something with this list once you've created and make an assert based on that?

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java (hunk 21 above)
Comment (abhandal) [reply app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: That's correct

File: app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java (hunk 22 above)
Comment (abhandal) [reply app/src/main/java/org/wikipedia/travel/MainPlannerFragment.java]: It updates the local arrayList with any changes that have been made to the destinaion history (e.g deletes, creates).

File: app/src/androidTest/java/org/wikipedia/espresso/travel/DestinationHistoryEspressoTest.java [hunk 25]
```diff
@@ -0,0 +1,86 @@
...
+        ViewInteraction destinationInput = onView(withId(R.id.fragment_place_autocomplete));
+
+        destinationInput.perform(click());
+
+        UiObject searchBar = mDevice.findObject(new UiSelector()
+                .className("android.widget.EditText"));
+
+        try {
+            searchBar.setText("Montreal");
+            closeSoftKeyboard();
```
Comment (sophiaquach16) [inline app/src/androidTest/java/org/wikipedia/espresso/travel/DestinationHistoryEspressoTest.java]: What is closeSoftKeyboard()?

Comment (abhandal) [reply app/src/androidTest/java/org/wikipedia/espresso/travel/DestinationHistoryEspressoTest.java]: It closes the keyboard after the user enters "Montreal" into the search field.

File: app/src/test/java/org/wikipedia/travel/database/DestinationDbHelperTest.java (hunk 24 above)
Comment (abhandal) [reply app/src/test/java/org/wikipedia/travel/database/DestinationDbHelperTest.java]: Updated the unit test in the new commit

**Comment by amawai** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 30
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985&bid=7015725)

File: app/src/androidTest/java/org/wikipedia/travel/database/DestinationDbHelperTest.java [hunk 26]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.List;
+
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+public class DestinationDbHelperTest {
+
+    @Test
+    public void testCreateAndDeleteList() {
```
Comment (amawai) [inline app/src/androidTest/java/org/wikipedia/travel/database/DestinationDbHelperTest.java]: As it is a unit test, perhaps create separate tests for list creation and deletion 
Also, given the comments about last release's unit tests, is it possible to test edge cases such as the following deletion of list that doesn't exist in the db?

File: app/src/main/java/org/wikipedia/travel/trip/Trip.java [hunk 27]
```diff
@@ -17,16 +18,24 @@
...
-    @NonNull private String title;
-    @SuppressWarnings("unused") @Nullable private List<Destination> destinations;
-    @SuppressWarnings("unused") @Nullable private Destination singleDestination;
-    @SuppressWarnings("unused") @Nullable private Date departureDate;
-
-    //Initialization of database table for trips with single destinations
-    public static final TripDatabaseTable DATABASE_TABLE = new TripDatabaseTable();
+    @NonNull
+    private String title;
+    @SuppressWarnings("unused")
```
Comment (amawai) [inline app/src/main/java/org/wikipedia/travel/trip/Trip.java]: If these attributes are never used, is it possible to delete them?

Comment (abhandal) [reply app/src/main/java/org/wikipedia/travel/trip/Trip.java]: The attributes existed in the previous merge, it was moved to a new line. With that said, those tags are used in the code.

File: app/src/androidTest/java/org/wikipedia/travel/database/DestinationDbHelperTest.java (hunk 26 above)
Comment (abhandal) [reply app/src/androidTest/java/org/wikipedia/travel/database/DestinationDbHelperTest.java]: Separated the test into their own unit and added edge test cases.

**Comment by amawai** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 30
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/amawai/android-wikipedia-390/pullRequest?prid=1526985&bid=7015725)

File: app/src/androidTest/java/org/wikipedia/travel/database/DestinationDbHelperTest.java [hunk 28]
```diff
@@ -0,0 +1,95 @@
...
+import org.wikipedia.travel.trip.Trip;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static junit.framework.Assert.assertNotNull;
+import static junit.framework.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+public class DestinationDbHelperTest {
```
Comment (amawai) [inline app/src/androidTest/java/org/wikipedia/travel/database/DestinationDbHelperTest.java]: Solid and extensive test additions, it covers many edge cases.

**Comment by jadmalek** [comment]
**Acceptance Testing** 

These acceptance tests characterize the expected behavior the task was meant to exhibit. See explanation below for its exclusion from the enhancement of our trip planner feature (#26). 

| Test #         | Inputs | Expected Output  | Status |
| ------------------ | ------------- | ------------- | ------------- |
| 1. Plan a Trip card appears on application's main page.  | Wikipedia application launched.  | A card for the trip planner is displayed, along with the other cards, on the main page.  | PASS  |
| 2. Plan a Trip page display.  | Press on Plan a Trip arrow icon on card.  | The Plan a Trip page is depicted, including a list of trips and the option to plan a new trip.   | PASS  |
| 3. Destination picker display.  | Press on the "+ Plan a new trip" button. | Destinations page is depicted, including a search bar for a destination to be inputted.   | PASS  |
| 4. Particular destination selection.  | Type in the name of the destination and select on the dropdown menu, or simply click on the dropdown menu and select a desired destination. | Selected destination will appear in text format within the search bar.   | PASS  |
| 5. Destination trip history.  | In the Plan a Trip Page, select on trip that has previously been planned. | The planned trips destination that was searched for will appear under the search bar.  | PASS  |
| 6. Remove a destination from the search history.  | Select on the destination in the "Destination Search History" and swipe left. | The destination will be removed from the list of destinations in the selected trip. | PASS  |

**Demo Steps**

1. Launch the application.
2. Scroll down the main page until the Plan a Trip card is visible.
3. Press on the Plan a trip arrow icon to view the Plan a Trip screen.
4. Press on the "+ Plan a new trip" button in order to begin planning a new trip.
5. Search for a particular destination by typing in and selecting a destination provided by the search bar.
6. Go back to trip planner's main screen (no need to complete the entire plan for said trip).
7. Select on the edit ("pencil") icon of the trip planned in previous step.
8. Select on the destination in the "Destination Search History" and swipe left.
9. Confirm that destination has been removed from the germane history.
